tests/testthat/test_spatial_deviation_noise_matrix_rng.R

```diff
@@ -0,0 +1,67 @@
+test_that(
+  "Test if non-centered parameter transformation of MVN(0,I) agrees with stan
+  spatial_deviation_noise_matrix_rng.stan",
+  {
+    set.seed(2024)
+    model <- compiled_site_inf_model
+    space_model_fxns <- spatial_fxns$functions
+
+    corr_fun_params <- list(
+      dist_matrix = as.matrix(
+        dist(
+          data.frame(
+            x = c(85, 37, 48, 7),
+            y = c(12, 75, 81, 96),
+            diag = TRUE,
+            upper = TRUE
+          )
+        )
+      ),
+      phi = 25,
+      l = 1
+    )
+    omega_matrix <- space_model_fxns$exponential_decay_corr_func(
+      corr_fun_params$dist_matrix,
+      corr_fun_params$phi,
+      corr_fun_params$l
+    )
+    sigma_matrix <- sqrt(0.02) * omega_matrix
+    n_time <- 150
+
+    stan_noise_matrix <- space_model_fxns$spatial_deviation_noise_matrix_rng(
+      sigma_matrix,
+      n_time
+    )
+
+
+    # non-centered param
+    n_sites <- ncol(
+      corr_fun_params$dist_matrix
+    )
+    r_non_center_noise_matrix <- mvrnorm(
+      n = n_time,
+      mu = rep(0, n_sites),
+      Sigma = diag(n_sites)
+    )
+    sqrt_sigma_matrix <- expm::sqrtm(
+      sigma_matrix
+    )
+    r_noise_matrix <- apply(
+      t(r_non_center_noise_matrix),
+      2,
+      function(x) sqrt_sigma_matrix %*% x
+    )
+
+    # comparing with cramer test
+    cramer_p_value <- cramer.test(
+      stan_noise_matrix,
+      r_noise_matrix
+    )$p.value
+
+
+    testthat::expect_gt(
+      cramer_p_value,
```

Comment:
Can you explain to me like I am five how the cramer test works? A p-value of > 0.05 means that the two matrices are not statistically significantly different?

I would think you would want something like repeated across many simulations, generate the noise matrices both ways, and then compare the distributions of matrix elements to show they are similar enough/have the same statistical properties (so perhaps the same trace, the same determinant, etc)?

Reply:
@kaitejohnson the cramer test is a way to compare two multivariate samples, where the null hypothesis is that they are coming from the same distribution.  So, we "want" a high p-value, which gives us more confidence that these two multivariate samples come from the same distribution.  The test statistic it uses is in the paper I added in the spatial folder on sharepoint titled, `On a new multivariate two-sample test`, but this test statistic looks at the euclidean distance of two sampled vectors, with this I believe they make it easier to find the distribution of the test statistic to give us a way to get the p-value.

Reply:
@kaitejohnson I think it would be a good idea to do this across many simulations instead of just one seed, like I currently have it.  But I don't think looking at the trace or determinant would work since this noise matrix is `n_site x n_time`, which isn't typically a square matrix.  And if I am not mistaken, trace and determinant are only for square matrices.

Reply:
^You are definitely right about that :) Although for the test you could make them square matrices but probably best not to in order to keep it closest to the real situation

I might see if @athowes has thoughts here as this is somewhat related to unit testing when sampling from a target distribution. I just am sort of wary of relying on a p_value > 0.05 as a robust test to tell us that two draws are in fact coming from the same distribution, but yeah my brain breaks when we start to think about doing this across samples from matrices

Reply:
Don't fully understand context here but I have also used relying on p-value greater than 0.01 (rather than 0.05 but same idea) as a test. Summary of discussion was that moving the p-value closer to 0 and increasing sample size will reduce false negative rate.

And about the matrices, the point is that you have some distribution on spatio-temporal random effects, and conceptualise that as a distribution on matrices where one dimension is space and one is time? Seems possible. I wonder if there are any simpler tests on decompositions of that, but maybe you have those too already.

Reply:
From `?cramer.test` it looks like some kernel-based distance test. If it's a multivariate test, do you know that it's valid for matrices? I guess what you would be doing is putting the matrix into a vector (maybe all tests for vectors can be applied to matrices by doing that)

Reply:
@athowes this does bring an interesting point to using / trusting the cramer test.  I believe one of the assumptions is that observations of each sample are independent of each other.  In our case we know they are not, since we are looking at AR(1) on time differences, so I don't believe we can even use this test here.  But this test is valid for matrices that contain random vectors of some distribution, but after looking at the documentation a bit more, I need to switch how I am passing in the matrix into `cramer.test`.  

Here is the paper I looked at to see if I could use this test, just in case you don't have access to the spatial folder in sharepoint :
[On a new multivariate two-sample test.pdf](https://github.com/user-attachments/files/16472378/On.a.new.multivariate.two-sample.test.pdf)

@athowes and / or @kaitejohnson do you know of any multivariate tests that don't have the this independence assumption or should we not be concerned about it?

Reply:
I think an alternative might be to apply a test to just a single time point (so the resulting vector), since the point of this test is to ensure that the non-centered parameterization via the cholesky decomposition is equivalent to the original centered parameterization? 

That way, we could just generate $n$ vectors from each MVN and use the KS test for comparing the vectors? Does that make sense? Perhaps am over simplifying

Reply:
After a conversation with Kaitlyn, non-independence among random vectors will not be a problem since we are not looking at the $\vec{\delta}(t)$ term from the model in https://github.com/CDCgov/ww-inference-model/issues/28#issue-2414103263, we are looking at the noise term, or the $\vec{\epsilon}$ term.  So I believe it is fine to do the cramer test for this case, I just need to fix how we pass in the matrix of random vectors into the `cramer.test` function as it accepts a matrix with rows as random vectors and I currently have it set up as a matrix with columns as random vectors.